GooglePlayInstant/Editor/QuickDeploy/LoadingScreenGenerator.cs

```diff
@@ -234,22 +234,37 @@ private static Vector2 GetPreImportTextureDimensions(Texture2D texture)
             return new Vector2(tempTexture.width, tempTexture.height);
         }
 
-        // Visible for testing.
-        internal static Sprite FindReplayButtonSprite()
+        /// <summary>
+        /// Searches the AssetDatabase for an Asset of type T using the searchFilter string.
+        /// See https://docs.unity3d.com/ScriptReference/AssetDatabase.FindAssets.html for what can be included in that
+        /// string.
+        /// </summary>
+        /// <returns>The first found asset or null if no assets are found.</returns>
+        public static T FindAssetByFilter<T>(string searchFilter) where T : UnityEngine.Object
         {
-            // We search for the sprite by name instead of by its path, because we don't require developers to keep the
+            // We search for the asset by name instead of by its path, because we don't require developers to keep the
             // play-instant-unity-plugin folder directly inside the Assets folder.
-            string searchFilter = "GooglePlayInstantRetryButton t:sprite";
             string[] foundGuids = AssetDatabase.FindAssets(searchFilter);
 
-            if (foundGuids.Length <= 0)
+            if (foundGuids.Length == 0)
             {
-                Debug.LogError("Failed to obtain retry button texture.");
+                Debug.LogErrorFormat("Failed to obtain asset from filter: {0}", searchFilter);
                 return null;
             }
 
+            if (foundGuids.Length > 1)
+            {
+                Debug.LogErrorFormat("Found multiple assets that match: {0}", searchFilter);
+            }
+
             string path = AssetDatabase.GUIDToAssetPath(foundGuids[0]);
-            return AssetDatabase.LoadAssetAtPath<Sprite>(path);
+            return AssetDatabase.LoadAssetAtPath<T>(path);
+        }
+
+        // Visible for testing.
+        internal static Sprite FindReplayButtonSprite()
+        {
+            return FindAssetByFilter<Sprite>("GooglePlayInstantRetryButton t:sprite");
```

Comment:
It seems like if the replay button sprite is missing, that's a major usability problem.